### ISSUE TITLE
Merging SideBarItemConfig with MainItemConfig

### ISSUE THREAD

**Issue by patrickacollier** [matt8707/ha-fusion#535]
@matt8707 curious what your thoughts are, or how you thought about handling the components that are only available to sidebar. Do you think SidebarItemConfig and MainItemConfig should be refactored into 1?
If separate, how would you imagine handling the content rendering? 

I've created a PR  [(on my local)](https://github.com/patrickacollier/ha-fusion/pull/1 ) just to see what all it would take (this is keeping both Main and Sidebar config). I've only implemented "Bar" from sidebar but all objects are available and editable. But determining the best way to display the content as they are implemented differently. 

Also this is my first svelte project, so just a few hours in. open for any suggestions.
![image](https://github.com/user-attachments/assets/d8ebe699-0148-4bb2-83e8-0d1f5deb1e28)
*also, with this test you can't actually edit the component once its created likely due to the differences in sidebar/main implements.

**Comment by matt8707** [comment]
Yeah, there will be small (or large) differences in how the component looks and behaves depending on whether it's in the sidebar or main section. To stay sane while prototyping, they're split up for now. However, the plan is to merge them further down the line.
